axlearn/common/trainer.py

```diff
@@ -566,6 +575,11 @@ def run(
             self.mesh(),
             jax.log_compiles(self.vlog_is_on(1)),
             self._context_manager(),
+            (
+                self._recorder.maybe_monitor_all_goodput()
```

Comment:
Should this be more general, instead of only about godput? Since `recorder` is not necessarily a goodput recorder?

Reply:
Changed it to `maybe_monitor_all`.